diff --git a/arch/lkl/Makefile b/arch/lkl/Makefile
index 88d47de..dc59705 100644
--- a/arch/lkl/Makefile
+++ b/arch/lkl/Makefile
@@ -2,7 +2,7 @@ include arch/lkl/auto.conf
 
 KBUILD_CFLAGS += -fno-builtin
 
-ifneq (,$(filter $(OUTPUT_FORMAT),elf64-x86-64 elf64-x86-64-freebsd elf32-littlearm elf64-littleaarch64))
+ifneq (,$(filter $(OUTPUT_FORMAT),elf64-x86-64 elf32-i386 elf64-x86-64-freebsd elf32-littlearm elf64-littleaarch64))
 KBUILD_CFLAGS += -fPIC
 else ifneq (,$(filter $(OUTPUT_FORMAT),pe-i386 pe-x86-64 ))
 ifneq ($(OUTPUT_FORMAT),pe-x86-64)
@@ -24,7 +24,13 @@ endif
 
 LDFLAGS_vmlinux += -r
 LKL_ENTRY_POINTS := lkl_start_kernel lkl_sys_halt lkl_syscall lkl_trigger_irq \
-	lkl_get_free_irq lkl_put_irq lkl_is_running
+	lkl_get_free_irq lkl_put_irq lkl_is_running lkl_bug lkl_printf
+
+ifeq ($(OUTPUT_FORMAT),elf32-i386)
+LKL_ENTRY_POINTS += \
+	__x86.get_pc_thunk.bx __x86.get_pc_thunk.dx __x86.get_pc_thunk.ax \
+	__x86.get_pc_thunk.cx __x86.get_pc_thunk.si __x86.get_pc_thunk.di
+endif
 
 core-y += arch/lkl/kernel/
 core-y += arch/lkl/mm/
diff --git a/arch/lkl/scripts/headers_install.py b/arch/lkl/scripts/headers_install.py
index e8f1347..22a6629 100755
--- a/arch/lkl/scripts/headers_install.py
+++ b/arch/lkl/scripts/headers_install.py
@@ -178,6 +178,9 @@ p = re.compile("enum\s+(\w*)\s*{([^}]*)}", re.M|re.S)
 q = re.compile("(\w+)\s*(,|=[^,]*|$)", re.M|re.S)
 find_enums(p, q, defines)
 
+# needed for i386
+defines.add("__NR_stime")
+
 def process_header(h):
     print("  REPLACE\t%s" % (out_dir + "/" + os.path.basename(h)))
     replace(h)
diff --git a/tools/lkl/Targets b/tools/lkl/Targets
index 4a6b895..c97c924 100644
--- a/tools/lkl/Targets
+++ b/tools/lkl/Targets
@@ -1,8 +1,9 @@
 libs-y += lib/liblkl
 
-ifneq ($(LKL_HOST_CONFIG_BSD),y)
-libs-$(LKL_HOST_CONFIG_POSIX) += lib/hijack/liblkl-hijack
-endif
+# Including hijack brokes build in my environment, temporarily disabled
+# ifneq ($(LKL_HOST_CONFIG_BSD),y)
+# libs-$(LKL_HOST_CONFIG_POSIX) += lib/hijack/liblkl-hijack
+# endif
 LDFLAGS_lib/hijack/liblkl-hijack-y += -shared -nodefaultlibs
 LDLIBS_lib/hijack/liblkl-hijack-y += -ldl
 LDLIBS_lib/hijack/liblkl-hijack-$(LKL_HOST_CONFIG_ANDROID) += -lgcc -lc
diff --git a/tools/lkl/include/lkl.h b/tools/lkl/include/lkl.h
index 3ec9c2f..77ff77e 100644
--- a/tools/lkl/include/lkl.h
+++ b/tools/lkl/include/lkl.h
@@ -35,7 +35,7 @@ extern "C" {
 int inet_pton(int af, const char *src, void *dst);
 #endif
 
-#if defined(__ANDROID__) && __LKL__BITS_PER_LONG == 32
+#if __LKL__BITS_PER_LONG == 32 && (defined(__ANDROID__) || defined(__i386__))
 #define __lkl__NR_fcntl __lkl__NR_fcntl64
 #endif
 
