diff -aur acpica-R06_28_23/source/include/platform/acenv.h ../build/extbld/third_party/lib/acpica/acpica-R06_28_23/source/include/platform/acenv.h 
--- acpica-R06_28_23/source/include/platform/acenv.h    2023-06-29 00:11:26.000000000 +0300
+++ ../build/extbld/third_party/lib/acpica/acpica-R06_28_23/source/include/platform/acenv.h       2023-10-02 20:53:56.384424860 +0300
@@ -350,6 +350,9 @@
 #elif defined(__QNX__)
 #include "acqnx.h"
 
+#elif defined(__EMBOX__)
+#include "acembox.h"
+
 /*
  * EFI applications can be built with -nostdlib, in this case, it must be
  * included after including all other host environmental definitions, in