diff --color -aur rt-tests-2.9/src/cyclictest/cyclictest.c ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/cyclictest/cyclictest.c
--- rt-tests-2.9/src/cyclictest/cyclictest.c	2025-07-01 17:20:28.000000000 +0300
+++ ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/cyclictest/cyclictest.c	2026-01-27 10:36:05.715798017 +0300
@@ -52,7 +52,7 @@
 #define SCHED_NORMAL SCHED_OTHER
 #endif
 
-#define sigev_notify_thread_id _sigev_un._tid
+// #define sigev_notify_thread_id _sigev_un._tid
 
 #ifdef __UCLIBC__
 #define MAKE_PROCESS_CPUCLOCK(pid, clock) \
@@ -1039,8 +1039,8 @@
 {
 	if (strncasecmp(polname, "other", 5) == 0)
 		policy = SCHED_OTHER;
-	else if (strncasecmp(polname, "batch", 5) == 0)
-		policy = SCHED_BATCH;
+	// else if (strncasecmp(polname, "batch", 5) == 0)
+	// 	policy = SCHED_BATCH;
 	else if (strncasecmp(polname, "idle", 4) == 0)
 		policy = SCHED_IDLE;
 	else if (strncasecmp(polname, "fifo", 4) == 0)
@@ -1065,9 +1065,9 @@
 	case SCHED_RR:
 		policystr = "rr";
 		break;
-	case SCHED_BATCH:
-		policystr = "batch";
-		break;
+	// case SCHED_BATCH:
+	// 	policystr = "batch";
+	// 	break;
 	case SCHED_IDLE:
 		policystr = "idle";
 		break;
diff --color -aur rt-tests-2.9/src/include/rt-utils.h ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/include/rt-utils.h
--- rt-tests-2.9/src/include/rt-utils.h	2025-07-01 17:20:28.000000000 +0300
+++ ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/include/rt-utils.h	2026-01-27 10:32:57.351436579 +0300
@@ -34,9 +34,17 @@
 void tracemark(char *fmt, ...) __attribute__((format(printf, 1, 2)));
 void disable_trace_mark(void);
 
-#define MSEC_PER_SEC		1000
-#define USEC_PER_SEC		1000000
-#define NSEC_PER_SEC		1000000000
+#define VERSION 2.9
+
+#ifndef MSEC_PER_SEC
+	#define MSEC_PER_SEC		1000
+#endif
+#ifndef USEC_PER_SEC
+	#define USEC_PER_SEC		1000000
+#endif
+#ifndef NSEC_PER_SEC
+	#define NSEC_PER_SEC		1000000000
+#endif
 #define USEC_TO_NSEC(u)		((u) * 1000)
 #define USEC_TO_SEC(u)		(u) / USEC_PER_SEC)
 #define NSEC_TO_USEC(n)		((n) / 1000)
diff --color -aur rt-tests-2.9/src/lib/rt-utils.c ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/lib/rt-utils.c
--- rt-tests-2.9/src/lib/rt-utils.c	2025-07-01 17:20:28.000000000 +0300
+++ ../build/extbld/third_party/rt_tests/src/rt-tests-2.9/src/lib/rt-utils.c	2026-01-27 10:32:57.351564729 +0300
@@ -303,10 +303,10 @@
 		return "SCHED_FIFO";
 	case SCHED_RR:
 		return "SCHED_RR";
-	case SCHED_BATCH:
-		return "SCHED_BATCH";
-	case SCHED_IDLE:
-		return "SCHED_IDLE";
+	// case SCHED_BATCH:
+	// 	return "SCHED_BATCH";
+	// case SCHED_IDLE:
+	// 	return "SCHED_IDLE";
 	case SCHED_DEADLINE:
 		return "SCHED_DEADLINE";
 	}
@@ -322,10 +322,10 @@
 		return SCHED_FIFO;
 	else if (!strcmp(str, "rr"))
 		return SCHED_RR;
-	else if (!strcmp(str, "batch"))
-		return SCHED_BATCH;
-	else if (!strcmp(str, "idle"))
-		return SCHED_IDLE;
+	// else if (!strcmp(str, "batch"))
+	// 	return SCHED_BATCH;
+	// else if (!strcmp(str, "idle"))
+	// 	return SCHED_IDLE;
 	else if (!strcmp(str, "deadline"))
 		return SCHED_DEADLINE;
 