diff -aur opencv-4.5.2/CMakeLists.txt ../build/extbld/third_party/lib/opencv/build/opencv-4.5.2/CMakeLists.txt
--- opencv-4.5.2/CMakeLists.txt	2021-04-02 14:23:54.000000000 +0300
+++ ../build/extbld/third_party/lib/opencv/build/opencv-4.5.2/CMakeLists.txt	2021-04-14 16:08:12.626665220 +0300
@@ -686,14 +686,11 @@
 #  Detect endianness of build platform
 # ----------------------------------------------------------------------------
 
-if(IOS)
-  # test_big_endian needs try_compile, which doesn't work for iOS
-  # http://public.kitware.com/Bug/view.php?id=12288
-  set(WORDS_BIGENDIAN 0)
-else()
-  include(TestBigEndian)
-  test_big_endian(WORDS_BIGENDIAN)
-endif()
+# For Embox we cannot run cmake's TestBigEndian properly, because
+# TestBigEndian creates ELF file. It requires a full linking with Embox,
+# and TestBigEndian's main() is discarded as it is not entry point.
+# https://github.com/embox/embox/issues/2419
+set(WORDS_BIGENDIAN 0)
 
 # ----------------------------------------------------------------------------
 #  Detect 3rd-party libraries
