diff -aur ./opencv-4.0.1/cmake/OpenCVFindLibsVideo.cmake ../build/extbld/third_party/lib/opencv/build/opencv-4.0.1/cmake/OpenCVFindLibsVideo.cmake
--- ./opencv-4.0.1/cmake/OpenCVFindLibsVideo.cmake  2018-12-22 10:03:30.000000000 +0300
+++ ../build/extbld/third_party/lib/opencv/build/opencv-4.0.1/cmake/OpenCVFindLibsVideo.cmake   2024-06-19 18:04:31.443739746 +0300
@@ -195,38 +195,39 @@
     else()
       message(STATUS "Can't find FFmpeg via find_package(${OPENCV_FFMPEG_USE_FIND_PACKAGE})")
     endif()
-  elseif(WIN32 AND NOT ARM AND NOT OPENCV_FFMPEG_SKIP_DOWNLOAD)
-    include("${OpenCV_SOURCE_DIR}/3rdparty/ffmpeg/ffmpeg.cmake")
-    download_win_ffmpeg(FFMPEG_CMAKE_SCRIPT)
-    if(FFMPEG_CMAKE_SCRIPT)
-      set(HAVE_FFMPEG TRUE)
-      set(HAVE_FFMPEG_WRAPPER 1)
-      include("${FFMPEG_CMAKE_SCRIPT}")
-    endif()
-  elseif(PKG_CONFIG_FOUND)
-    ocv_check_modules(FFMPEG libavcodec libavformat libavutil libswscale)
-    ocv_check_modules(FFMPEG_libavresample libavresample)
-    if(FFMPEG_libavresample_FOUND)
-      ocv_append_build_options(FFMPEG FFMPEG_libavresample)
-    endif()
+  elseif(DEFINED FFMPEG_INCLUDE_DIR AND DEFINED FFMPEG_LIB_DIR)
+    set(HAVE_FFMPEG TRUE)
+    set(FFMPEG_INCLUDE_DIRS ${FFMPEG_INCLUDE_DIR})
+    set(FFMPEG_LIBRARY_DIRS ${FFMPEG_LIB_DIR})
+    set(FFMPEG_LIBRARIES "${FFMPEG_LIB_DIR}/libavcodec.a;${FFMPEG_LIB_DIR}/libavformat.a;${FFMPEG_LIB_DIR}/libavutil.a;${FFMPEG_LIB_DIR}/libswscale.a")
+    set(FFMPEG_libavcodec_FOUND TRUE)
+    set(FFMPEG_libavformat_FOUND TRUE)
+    set(FFMPEG_libavutil_FOUND TRUE)
+    set(FFMPEG_libswscale_FOUND TRUE)
+    set(FFMPEG_libavcodec_VERSION 4.3.1)
+    set(FFMPEG_libavformat_VERSION 4.3.1)
+    set(FFMPEG_libavutil_VERSION 4.3.1)
+    set(FFMPEG_libswscale_VERSION 4.3.1)
+    message(STATUS "FFmpeg has been found with custom settings.")
   else()
-    message(STATUS "Can't find ffmpeg - 'pkg-config' utility is missing")
+    message(STATUS "Can't find FFmpeg")
   endif()
 endif()
-if(HAVE_FFMPEG
-    AND NOT HAVE_FFMPEG_WRAPPER
-)
+
+if(HAVE_FFMPEG AND NOT HAVE_FFMPEG_WRAPPER)
+  string(REPLACE "-Wl,--gc-sections" "" CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}")
   try_compile(__VALID_FFMPEG
       "${OpenCV_BINARY_DIR}"
       "${OpenCV_SOURCE_DIR}/cmake/checks/ffmpeg_test.cpp"
       CMAKE_FLAGS "-DINCLUDE_DIRECTORIES:STRING=${FFMPEG_INCLUDE_DIRS}"
                   "-DLINK_DIRECTORIES:STRING=${FFMPEG_LIBRARY_DIRS}"
                   "-DLINK_LIBRARIES:STRING=${FFMPEG_LIBRARIES}"
+                  "-DCMAKE_EXE_LINKER_FLAGS:STRING=${CMAKE_EXE_LINKER_FLAGS}"
       OUTPUT_VARIABLE TRY_OUT
   )
   if(NOT __VALID_FFMPEG)
-    #message(FATAL_ERROR "FFMPEG: test check build log:\n${TRY_OUT}")
     message(STATUS "WARNING: Can't build ffmpeg test code")
+    message(STATUS "FFmpeg test build log:\n${TRY_OUT}")
     set(HAVE_FFMPEG FALSE)
   else()
     ocv_append_build_options(VIDEOIO FFMPEG)
diff -aur ./opencv-4.0.1/cmake/checks/ffmpeg_test.cpp ../build/extbld/third_party/lib/opencv/build/opencv-4.0.1/cmake/checks/ffmpeg_test.cpp
--- ./opencv-4.0.1/cmake/checks/ffmpeg_test.cpp 2018-12-22 10:03:30.000000000 +0300
+++ ../build/extbld/third_party/lib/opencv/build/opencv-4.0.1/cmake/checks/ffmpeg_test.cpp  2024-06-19 17:57:21.792792506 +0300
@@ -1,4 +1,5 @@
 #define __STDC_CONSTANT_MACROS
+#define __STDCPP_WANT_MATH_SPEC_FUNCS__ 1
 
 #include <stdlib.h>
 
