diff --color -aur ./ardupilot-Copter-4.6.2/libraries/AP_Param/AP_Param.cpp ../build/extbld/project/ardupilot/sitl/ardupilot-Copter-4.6.2/libraries/AP_Param/AP_Param.cpp
--- ./ardupilot-Copter-4.6.2/libraries/AP_Param/AP_Param.cpp    2025-07-18 02:43:00.000000000 +0300
+++ ../build/extbld/project/ardupilot/sitl/ardupilot-Copter-4.6.2/libraries/AP_Param/AP_Param.cpp       2025-10-08 21:12:39.989277571 +0300
@@ -24,7 +24,7 @@
 #include "AP_Param.h"
 
 #include <cmath>
-#include <string.h>
+#include <strings.h>
 
 #include <AP_Common/AP_Common.h>
 #include <AP_HAL/AP_HAL.h>
diff --color -aur ./ardupilot-Copter-4.6.2/Tools/ardupilotwaf/boards.py ../build/extbld/project/ardupilot/sitl/ardupilot-Copter-4.6.2/Tools/ardupilotwaf/boards.py
--- ./ardupilot-Copter-4.6.2/Tools/ardupilotwaf/boards.py       2025-07-18 02:43:00.000000000 +0300
+++ ../build/extbld/project/ardupilot/sitl/ardupilot-Copter-4.6.2/Tools/ardupilotwaf/boards.py  2025-10-08 21:16:36.417536457 +0300
@@ -246,17 +246,17 @@
             '-Wno-redundant-decls',
             '-Wno-unknown-pragmas',
             '-Wno-trigraphs',
-            '-Werror=shadow',
+            '-Wno-error=shadow',
             '-Werror=return-type',
             '-Werror=unused-result',
             '-Werror=unused-variable',
             '-Werror=narrowing',
             '-Werror=attributes',
             '-Werror=overflow',
-            '-Werror=parentheses',
+            '-Wno-error=parentheses',
             '-Werror=format-extra-args',
             '-Werror=ignored-qualifiers',
-            '-Werror=undef',
+            '-Wno-error=undef',
             '-DARDUPILOT_BUILD',
         ]
 
@@ -379,15 +379,15 @@
             '-Werror=switch',
             '-Werror=sign-compare',
             '-Werror=type-limits',
-            '-Werror=undef',
+            '-Wno-error=undef',
             '-Werror=unused-result',
-            '-Werror=shadow',
+            '-Wno-error=shadow',
             '-Werror=unused-value',
             '-Werror=unused-variable',
             '-Werror=delete-non-virtual-dtor',
             '-Wfatal-errors',
             '-Wno-trigraphs',
-            '-Werror=parentheses',
+            '-Wno-error=parentheses',
             '-DARDUPILOT_BUILD',
             '-Wuninitialized',
             '-Warray-bounds',
@@ -760,7 +760,7 @@
         cfg.check_librt(env)
         cfg.check_feenableexcept()
 
-        env.LINKFLAGS += ['-pthread',]
+        # env.LINKFLAGS += ['-pthread',]
 
         if cfg.env.DEBUG and 'clang++' in cfg.env.COMPILER_CXX and cfg.options.asan:
              env.LINKFLAGS += ['-fsanitize=address']
