diff -aur ./k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h ../build/extbld/third_party/bsp/niiet/k1921vk035_system/k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h
--- ./k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h	2022-03-06 13:57:50.000000000 +0300
+++ ../build/extbld/third_party/bsp/niiet/k1921vk035_system/k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h	2024-05-15 20:02:40.044919722 +0300
@@ -27,6 +27,14 @@
 
 //-- Includes ------------------------------------------------------------------
 #include <stdint.h>
+#ifdef __EMBOX__
+#include <framework/mod/options.h>
+#include <module/embox/arch/system.h>
+#include <module/third_party/bsp/niiet/k1921vk035_system.h>
+
+#define OSECLK_VAL OPTION_MODULE_GET(third_party__bsp__niiet__k1921vk035_system,NUMBER,external_clk_freq)
+#define SYSCLK_PLL
+#endif
 
 //-- Defines -------------------------------------------------------------------
 #define OSICLK_VAL 8000000
