diff -aur ./thrift-0.11.0/lib/cpp/src/thrift/protocol/TJSONProtocol.cpp ../build/extbld/third_party/lib/thrift/thrift-0.11.0/lib/cpp/src/thrift/protocol/TJSONProtocol.cpp
--- ./thrift-0.11.0/lib/cpp/src/thrift/protocol/TJSONProtocol.cpp	2017-12-03 22:11:36.000000000 +0300
+++ ../build/extbld/third_party/lib/thrift/thrift-0.11.0/lib/cpp/src/thrift/protocol/TJSONProtocol.cpp	2024-05-10 23:25:25.578410602 +0300
@@ -539,9 +539,9 @@
   std::string val;
 
   bool special = false;
-  switch (boost::math::fpclassify(num)) {
+  switch (1) {
   case FP_INFINITE:
-    if (boost::math::signbit(num)) {
+    if (1) {
       val = kThriftNegativeInfinity;
     } else {
       val = kThriftInfinity;
diff -aur ./thrift-0.11.0/lib/cpp/src/thrift/transport/THttpServer.cpp ../build/extbld/third_party/lib/thrift/thrift-0.11.0/lib/cpp/src/thrift/transport/THttpServer.cpp
--- ./thrift-0.11.0/lib/cpp/src/thrift/transport/THttpServer.cpp	2017-12-03 22:11:36.000000000 +0300
+++ ../build/extbld/third_party/lib/thrift/thrift-0.11.0/lib/cpp/src/thrift/transport/THttpServer.cpp	2024-05-10 23:26:51.282829590 +0300
@@ -20,6 +20,7 @@
 #include <cstdlib>
 #include <sstream>
 #include <iostream>
+#include <strings.h>
 
 #include <thrift/config.h>
 #include <thrift/transport/THttpServer.h>