--- lib60870-2.3.1/lib60870-C/src/hal/serial/linux/serial_port_linux.c	2022-03-11 21:15:52.000000000 +0300
+++ ../build/extbld/third_party/lib/libiec60870/lib60870-2.3.1/lib60870-C/src/hal/serial/linux/serial_port_linux.c	2023-12-30 06:28:48.682363654 +0300
@@ -257,7 +257,7 @@
 
     ssize_t result = write(self->fd, buffer + startPos, bufSize);
 
-    tcdrain(self->fd);
+    /*tcdrain(self->fd);*/
 
     self->lastSentTime = Hal_getTimeInMs();
 
