diff -aur mosquitto-2.0.15/client/Makefile ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/client/Makefile
--- mosquitto-2.0.15/client/Makefile	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/client/Makefile	2023-04-20 18:27:43.951892738 +0300
@@ -14,19 +14,19 @@
 endif
 endif
 
-all : ${ALL_DEPS}
+all : ${ALL_DEPS} ../lib/libmosquitto.a
 
 static : static_pub static_sub static_rr
 	# This makes mosquitto_pub/sub/rr versions that are statically linked with
 	# libmosquitto only.
 
-static_pub : pub_client.o pub_shared.o client_props.o client_shared.o ../lib/libmosquitto.a
+static_pub : pub_client.o pub_shared.o client_props.o client_shared.o
 	${CROSS_COMPILE}${CC} $^ -o mosquitto_pub ${CLIENT_LDFLAGS} ${STATIC_LIB_DEPS} ${CLIENT_STATIC_LDADD}
 
-static_sub : sub_client.o sub_client_output.o client_props.o client_shared.o ../lib/libmosquitto.a
+static_sub : sub_client.o sub_client_output.o client_props.o client_shared.o
 	${CROSS_COMPILE}${CC} $^ -o mosquitto_sub ${CLIENT_LDFLAGS} ${STATIC_LIB_DEPS} ${CLIENT_STATIC_LDADD}
 
-static_rr : rr_client.o client_props.o client_shared.o pub_shared.o sub_client_output.o ../lib/libmosquitto.a
+static_rr : rr_client.o client_props.o client_shared.o pub_shared.o sub_client_output.o
 	${CROSS_COMPILE}${CC} $^ -o mosquitto_rr ${CLIENT_LDFLAGS} ${STATIC_LIB_DEPS} ${CLIENT_STATIC_LDADD}
 
 mosquitto_pub : pub_client.o pub_shared.o client_shared.o client_props.o
diff -aur mosquitto-2.0.15/config.mk ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/config.mk
--- mosquitto-2.0.15/config.mk	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/config.mk	2023-04-20 18:27:43.951892738 +0300
@@ -21,15 +21,15 @@
 # Disabling this will also mean that passwords must be stored in plain text. It
 # is strongly recommended that you only disable WITH_TLS if you are not using
 # password authentication at all.
-WITH_TLS:=yes
+# WITH_TLS:=yes
 
 # Comment out to disable TLS/PSK support in the broker and client. Requires
 # WITH_TLS=yes.
 # This must be disabled if using openssl < 1.0.
-WITH_TLS_PSK:=yes
+WITH_TLS_PSK:=no
 
 # Comment out to disable client threading support.
-WITH_THREADING:=yes
+WITH_THREADING:=no
 
 # Comment out to remove bridge support from the broker. This allow the broker
 # to connect to other brokers and subscribe/publish to topics. You probably
@@ -73,7 +73,7 @@
 WITH_EC:=yes
 
 # Build man page documentation by default.
-WITH_DOCS:=yes
+WITH_DOCS:=no
 
 # Build with client support for SOCK5 proxy.
 WITH_SOCKS:=yes
@@ -82,7 +82,7 @@
 WITH_STRIP:=no
 
 # Build static libraries
-WITH_STATIC_LIBRARIES:=no
+WITH_STATIC_LIBRARIES:=yes
 
 # Use this variable to add extra library dependencies when building the clients
 # with the static libmosquitto library. This may be required on some systems
@@ -90,13 +90,13 @@
 CLIENT_STATIC_LDADD:=
 
 # Build shared libraries
-WITH_SHARED_LIBRARIES:=yes
+WITH_SHARED_LIBRARIES:=no
 
 # Build with async dns lookup support for bridges (temporary). Requires glibc.
 #WITH_ADNS:=yes
 
 # Build with epoll support.
-WITH_EPOLL:=yes
+WITH_EPOLL:=no
 
 # Build with bundled uthash.h
 WITH_BUNDLED_DEPS:=yes
@@ -108,7 +108,7 @@
 WITH_UNIX_SOCKETS:=yes
 
 # Build mosquitto_sub with cJSON support
-WITH_CJSON:=yes
+WITH_CJSON:=no
 
 # Build mosquitto with support for the $CONTROL topics.
 WITH_CONTROL:=yes
@@ -181,18 +181,18 @@
 PLUGIN_LDFLAGS:=$(LDFLAGS)
 
 ifneq ($(or $(findstring $(UNAME),FreeBSD), $(findstring $(UNAME),OpenBSD), $(findstring $(UNAME),NetBSD)),)
-	BROKER_LDADD:=$(BROKER_LDADD) -lm
+	BROKER_LDADD:=$(BROKER_LDADD) 
 	BROKER_LDFLAGS:=$(BROKER_LDFLAGS) -Wl,--dynamic-list=linker.syms
 	SEDINPLACE:=-i ""
 else
-	BROKER_LDADD:=$(BROKER_LDADD) -ldl -lm
+	BROKER_LDADD:=$(BROKER_LDADD)  
 	SEDINPLACE:=-i
 endif
 
 ifeq ($(UNAME),Linux)
-	BROKER_LDADD:=$(BROKER_LDADD) -lrt
+	BROKER_LDADD:=$(BROKER_LDADD) 
 	BROKER_LDFLAGS:=$(BROKER_LDFLAGS) -Wl,--dynamic-list=linker.syms
-	LIB_LIBADD:=$(LIB_LIBADD) -lrt
+	LIB_LIBADD:=$(LIB_LIBADD) 
 endif
 
 ifeq ($(WITH_SHARED_LIBRARIES),yes)
diff -aur mosquitto-2.0.15/lib/loop.c ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/lib/loop.c
--- mosquitto-2.0.15/lib/loop.c	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/lib/loop.c	2023-04-20 18:27:43.951892738 +0300
@@ -32,10 +32,6 @@
 #include "tls_mosq.h"
 #include "util_mosq.h"
 
-#if !defined(WIN32) && !defined(__SYMBIAN32__) && !defined(__QNX__)
-#define HAVE_PSELECT
-#endif
-
 int mosquitto_loop(struct mosquitto *mosq, int timeout, int max_packets)
 {
 #ifdef HAVE_PSELECT
diff -aur mosquitto-2.0.15/mosquitto.conf ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/mosquitto.conf
--- mosquitto-2.0.15/mosquitto.conf	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/mosquitto.conf	2023-04-20 18:27:43.955892782 +0300
@@ -244,7 +244,7 @@
 # force the listener to only use IPv6. If you want support for both IPv4 and
 # IPv6, then do not use the socket_domain option.
 #
-#socket_domain
+socket_domain ipv4
 
 # Bind the listener to a specific interface. This is similar to
 # the [ip address/host name] part of the listener definition, but is useful
diff -aur mosquitto-2.0.15/plugins/auth-by-ip/Makefile ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/auth-by-ip/Makefile
--- mosquitto-2.0.15/plugins/auth-by-ip/Makefile	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/auth-by-ip/Makefile	2023-04-20 18:27:43.955892782 +0300
@@ -1,27 +1,12 @@
 include ../../config.mk
 
-.PHONY : all binary check clean reallyclean test install uninstall
+.PHONY : all binary
 
 PLUGIN_NAME=mosquitto_auth_by_ip
 
 all : binary
 
-binary : ${PLUGIN_NAME}.so
+binary : ${PLUGIN_NAME}.o
 
-${PLUGIN_NAME}.so : ${PLUGIN_NAME}.c
-	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) -fPIC -shared $< -o $@
-
-reallyclean : clean
-clean:
-	-rm -f *.o ${PLUGIN_NAME}.so *.gcda *.gcno
-
-check: test
-test:
-
-install: ${PLUGIN_NAME}.so
-	# Don't install, these are examples only.
-	#$(INSTALL) -d "${DESTDIR}$(libdir)"
-	#$(INSTALL) ${STRIP_OPTS} ${PLUGIN_NAME}.so "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
-
-uninstall :
-	-rm -f "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
+${PLUGIN_NAME}.o : ${PLUGIN_NAME}.c
+	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) $< -o $@
diff -aur mosquitto-2.0.15/plugins/deny-protocol-version/Makefile ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/deny-protocol-version/Makefile
--- mosquitto-2.0.15/plugins/deny-protocol-version/Makefile	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/deny-protocol-version/Makefile	2023-04-20 18:27:43.955892782 +0300
@@ -8,22 +8,7 @@
 
 all : binary
 
-binary : ${PLUGIN_NAME}.so
+binary : ${PLUGIN_NAME}.o
 
-${PLUGIN_NAME}.so : ${PLUGIN_NAME}.c
-	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) -fPIC -shared $< -o $@
-
-reallyclean : clean
-clean:
-	-rm -f *.o ${PLUGIN_NAME}.so *.gcda *.gcno
-
-check: test
-test:
-
-install: ${PLUGIN_NAME}.so
-	# Don't install, these are examples only.
-	#$(INSTALL) -d "${DESTDIR}$(libdir)"
-	#$(INSTALL) ${STRIP_OPTS} ${PLUGIN_NAME}.so "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
-
-uninstall :
-	-rm -f "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
+${PLUGIN_NAME}.o : ${PLUGIN_NAME}.c
+	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) $< -o $@
diff -aur mosquitto-2.0.15/plugins/message-timestamp/Makefile ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/message-timestamp/Makefile
--- mosquitto-2.0.15/plugins/message-timestamp/Makefile	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/message-timestamp/Makefile	2023-04-20 18:27:43.955892782 +0300
@@ -1,27 +1,12 @@
 include ../../config.mk
 
-.PHONY : all binary check clean reallyclean test install uninstall
+.PHONY : all binary
 
 PLUGIN_NAME=mosquitto_message_timestamp
 
 all : binary
 
-binary : ${PLUGIN_NAME}.so
+binary : ${PLUGIN_NAME}.o
 
-${PLUGIN_NAME}.so : ${PLUGIN_NAME}.c
-	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) -fPIC -shared $< -o $@
-
-reallyclean : clean
-clean:
-	-rm -f *.o ${PLUGIN_NAME}.so *.gcda *.gcno
-
-check: test
-test:
-
-install: ${PLUGIN_NAME}.so
-	# Don't install, these are examples only.
-	#$(INSTALL) -d "${DESTDIR}$(libdir)"
-	#$(INSTALL) ${STRIP_OPTS} ${PLUGIN_NAME}.so "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
-
-uninstall :
-	-rm -f "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
+${PLUGIN_NAME}.o : ${PLUGIN_NAME}.c
+	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) $< -o $@
diff -aur mosquitto-2.0.15/plugins/payload-modification/Makefile ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/payload-modification/Makefile
--- mosquitto-2.0.15/plugins/payload-modification/Makefile	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/plugins/payload-modification/Makefile	2023-04-20 18:27:43.955892782 +0300
@@ -1,27 +1,12 @@
 include ../../config.mk
 
-.PHONY : all binary check clean reallyclean test install uninstall
+.PHONY : all binary
 
 PLUGIN_NAME=mosquitto_payload_modification
 
 all : binary
 
-binary : ${PLUGIN_NAME}.so
+binary : ${PLUGIN_NAME}.o
 
-${PLUGIN_NAME}.so : ${PLUGIN_NAME}.c
-	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) -fPIC -shared $< -o $@
-
-reallyclean : clean
-clean:
-	-rm -f *.o ${PLUGIN_NAME}.so *.gcda *.gcno
-
-check: test
-test:
-
-install: ${PLUGIN_NAME}.so
-	# Don't install, these are examples only.
-	#$(INSTALL) -d "${DESTDIR}$(libdir)"
-	#$(INSTALL) ${STRIP_OPTS} ${PLUGIN_NAME}.so "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
-
-uninstall :
-	-rm -f "${DESTDIR}${libdir}/${PLUGIN_NAME}.so"
+${PLUGIN_NAME}.o : ${PLUGIN_NAME}.c
+	$(CROSS_COMPILE)$(CC) $(PLUGIN_CPPFLAGS) $(PLUGIN_CFLAGS) $(PLUGIN_LDFLAGS) $< -o $@
diff -aur mosquitto-2.0.15/src/conf_includedir.c ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/src/conf_includedir.c
--- mosquitto-2.0.15/src/conf_includedir.c	2022-08-16 16:32:59.000000000 +0300
+++ ../build/extbld/third_party/mosquitto_build/mosquitto-2.0.15/src/conf_includedir.c	2023-04-20 18:27:43.955892782 +0300
@@ -42,7 +42,7 @@
 #if defined(__HAIKU__)
 #  include <syslog.h>
 #elif !defined(WIN32) && !defined(__CYGWIN__) && !defined(__QNX__)
-#  include <sys/syslog.h>
+#  include <syslog.h>
 #endif
 
 #include "mosquitto_broker_internal.h"
