diff -aur stm32f3discovery.orig/BSP/Components/l3gd20/l3gd20.h stm32f3discovery/BSP/Components/l3gd20/l3gd20.h
--- stm32f3discovery.orig/BSP/Components/l3gd20/l3gd20.h	2015-05-19 02:19:54.615088563 +0300
+++ stm32f3discovery/BSP/Components/l3gd20/l3gd20.h	2015-05-19 12:01:42.815331680 +0300
@@ -45,7 +45,7 @@
 #endif
 
 /* Includes ------------------------------------------------------------------*/
-#include "..\Common\gyro.h"
+#include "../Common/gyro.h"
 
 /** @addtogroup BSP
   * @{
diff -aur stm32f3discovery.orig/BSP/Components/lsm303dlhc/lsm303dlhc.h stm32f3discovery/BSP/Components/lsm303dlhc/lsm303dlhc.h
--- stm32f3discovery.orig/BSP/Components/lsm303dlhc/lsm303dlhc.h	2015-05-19 02:19:54.635088563 +0300
+++ stm32f3discovery/BSP/Components/lsm303dlhc/lsm303dlhc.h	2015-05-19 02:36:39.591081148 +0300
@@ -46,7 +46,7 @@
 #endif
 
 /* Includes ------------------------------------------------------------------*/
-#include "..\Common\accelero.h"
+#include "../Common/accelero.h"
 
 /** @addtogroup LSM303DLHC
   * @{
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h	2015-05-19 02:19:54.595088563 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h	2015-05-19 02:36:20.191081292 +0300
@@ -48,7 +48,7 @@
 /* Includes ------------------------------------------------------------------*/
 #include "stm32f3_discovery.h"
 /* Include Gyroscope component driver */
-#include "..\Components\lsm303dlhc\lsm303dlhc.h"
+#include "../Components/lsm303dlhc/lsm303dlhc.h"
    
 /** @addtogroup BSP
   * @{
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h	2015-05-19 02:19:54.595088563 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h	2015-05-19 02:20:37.095088249 +0300
@@ -48,7 +48,7 @@
 /* Includes ------------------------------------------------------------------*/
 #include "stm32f3_discovery.h"
 /* Include Gyroscope component driver */
-#include "..\Components\l3gd20\l3gd20.h"   
+#include "../Components/l3gd20/l3gd20.h"   
 
 /** @addtogroup BSP
   * @{
