diff -aur stm32f3discovery.orig/BSP/Components/l3gd20/l3gd20.h stm32f3discovery/BSP/Components/l3gd20/l3gd20.h
--- stm32f3discovery.orig/BSP/Components/l3gd20/l3gd20.h	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/Components/l3gd20/l3gd20.h	2015-06-09 17:08:48.889240280 +0300
@@ -45,7 +45,7 @@
 #endif
 
 /* Includes ------------------------------------------------------------------*/
-#include "..\Common\gyro.h"
+#include "../Common/gyro.h"
 
 /** @addtogroup BSP
   * @{
diff -aur stm32f3discovery.orig/BSP/Components/lsm303dlhc/lsm303dlhc.h stm32f3discovery/BSP/Components/lsm303dlhc/lsm303dlhc.h
--- stm32f3discovery.orig/BSP/Components/lsm303dlhc/lsm303dlhc.h	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/Components/lsm303dlhc/lsm303dlhc.h	2015-06-09 17:08:48.889240280 +0300
@@ -46,7 +46,7 @@
 #endif
 
 /* Includes ------------------------------------------------------------------*/
-#include "..\Common\accelero.h"
+#include "../Common/accelero.h"
 
 /** @addtogroup LSM303DLHC
   * @{
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.c stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.c
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.c	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.c	2015-06-09 17:08:48.889240280 +0300
@@ -117,7 +117,7 @@
   /* MEMS configuration ------------------------------------------------------*/
    /* Fill the accelerometer structure */
     LSM303DLHC_InitStructure.Power_Mode = LSM303DLHC_NORMAL_MODE;
-    LSM303DLHC_InitStructure.AccOutput_DataRate = LSM303DLHC_ODR_50_HZ;
+    LSM303DLHC_InitStructure.AccOutput_DataRate = LSM303DLHC_ODR_1344_HZ;
     LSM303DLHC_InitStructure.Axes_Enable= LSM303DLHC_AXES_ENABLE;
     LSM303DLHC_InitStructure.AccFull_Scale = LSM303DLHC_FULLSCALE_2G;
     LSM303DLHC_InitStructure.BlockData_Update = LSM303DLHC_BlockUpdate_Continous;
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_accelerometer.h	2015-06-09 17:08:48.889240280 +0300
@@ -48,7 +48,7 @@
 /* Includes ------------------------------------------------------------------*/
 #include "stm32f3_discovery.h"
 /* Include Gyroscope component driver */
-#include "..\Components\lsm303dlhc\lsm303dlhc.h"
+#include "../Components/lsm303dlhc/lsm303dlhc.h"
    
 /** @addtogroup BSP
   * @{
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.c stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.c
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.c	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.c	2015-06-09 17:11:16.313245639 +0300
@@ -115,7 +115,7 @@
 
     /* Configure Mems : data rate, power mode, full scale and axes */
     L3GD20_InitStructure.Power_Mode = L3GD20_MODE_ACTIVE;
-    L3GD20_InitStructure.Output_DataRate = L3GD20_OUTPUT_DATARATE_1;
+    L3GD20_InitStructure.Output_DataRate = L3GD20_OUTPUT_DATARATE_4;
     L3GD20_InitStructure.Axes_Enable = L3GD20_AXES_ENABLE;
     L3GD20_InitStructure.Band_Width = L3GD20_BANDWIDTH_4;
     L3GD20_InitStructure.BlockData_Update = L3GD20_BlockDataUpdate_Continous;
diff -aur stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h
--- stm32f3discovery.orig/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h	2015-06-09 17:10:16.081243449 +0300
+++ stm32f3discovery/BSP/STM32F3-Discovery/stm32f3_discovery_gyroscope.h	2015-06-09 17:08:48.893240280 +0300
@@ -48,7 +48,7 @@
 /* Includes ------------------------------------------------------------------*/
 #include "stm32f3_discovery.h"
 /* Include Gyroscope component driver */
-#include "..\Components\l3gd20\l3gd20.h"   
+#include "../Components/l3gd20/l3gd20.h"   
 
 /** @addtogroup BSP
   * @{
