diff -aur pjproject-2.2.1.orig/pjlib/src/pj/os_core_unix.c pjproject-2.2.1/pjlib/src/pj/os_core_unix.c
--- pjproject-2.2.1.orig/pjlib/src/pj/os_core_unix.c	2014-08-21 16:11:18.346112334 +0400
+++ pjproject-2.2.1/pjlib/src/pj/os_core_unix.c	2014-08-21 15:40:38.959499883 +0400
@@ -994,7 +994,8 @@
 PJ_DEF(pj_status_t) pj_thread_local_alloc(long *p_index)
 {
 #if PJ_HAS_THREADS
-    pthread_key_t key;
+    static int key_idx = 0;
+    pthread_key_t key = key_idx++;
     int rc;
 
     PJ_ASSERT_RETURN(p_index != NULL, PJ_EINVAL);
diff -aur pjproject-2.2.1.orig/pjlib/src/pj/sock_bsd.c pjproject-2.2.1/pjlib/src/pj/sock_bsd.c
--- pjproject-2.2.1.orig/pjlib/src/pj/sock_bsd.c	2014-08-21 16:11:18.346112334 +0400
+++ pjproject-2.2.1/pjlib/src/pj/sock_bsd.c	2014-08-21 14:58:58.112906927 +0400
@@ -52,7 +52,11 @@
 const pj_uint16_t PJ_SOCK_STREAM= SOCK_STREAM;
 const pj_uint16_t PJ_SOCK_DGRAM	= SOCK_DGRAM;
 const pj_uint16_t PJ_SOCK_RAW	= SOCK_RAW;
+#if defined(SOCK_RDM)
 const pj_uint16_t PJ_SOCK_RDM	= SOCK_RDM;
+#else
+const pj_uint16_t PJ_SOCK_RDM	= 0xFFFF;
+#endif
 
 /*
  * Socket level values.
--- ../../../../../download/pjproject-2.2.1/build.mak.in	2014-02-21 11:28:39.000000000 +0400
+++ pjproject-2.2.1/build.mak.in	2014-10-02 17:00:36.911428486 +0400
@@ -109,7 +109,7 @@
 ifneq ($(findstring pa,@ac_pjmedia_snd@),)
 ifeq (@ac_external_pa@,1)
 # External PA
-APP_THIRD_PARTY_EXT += -lportaudio
+#APP_THIRD_PARTY_EXT += -lportaudio
 else
 APP_THIRD_PARTY_LIB_FILES += $(PJ_DIR)/third_party/lib/libportaudio-$(LIB_SUFFIX)
 ifeq ($(PJ_SHARED_LIBRARIES),)
--- ../../../../../download/pjproject-2.2.1/pjlib/include/pj/config.h	2014-03-14 10:58:10.000000000 +0400
+++ pjproject-2.2.1/pjlib/include/pj/config.h	2014-10-06 18:11:23.231784157 +0400
@@ -581,7 +581,7 @@
  * the underlying implementation changes.
  */
 #ifndef PJ_IOQUEUE_MAX_HANDLES
-#   define PJ_IOQUEUE_MAX_HANDLES	(64)
+#   define PJ_IOQUEUE_MAX_HANDLES	(8)
 #endif
 
 
