diff -aur pjproject-2.2.1.orig/build.mak.in pjproject-2.2.1/build.mak.in
--- pjproject-2.2.1.orig/build.mak.in	2015-01-12 02:52:08.115341907 +0300
+++ pjproject-2.2.1/build.mak.in	2015-01-12 02:51:49.515340847 +0300
@@ -109,7 +109,7 @@
 ifneq ($(findstring pa,@ac_pjmedia_snd@),)
 ifeq (@ac_external_pa@,1)
 # External PA
-APP_THIRD_PARTY_EXT += -lportaudio
+#APP_THIRD_PARTY_EXT += -lportaudio
 else
 APP_THIRD_PARTY_LIB_FILES += $(PJ_DIR)/third_party/lib/libportaudio-$(LIB_SUFFIX)
 ifeq ($(PJ_SHARED_LIBRARIES),)
@@ -254,7 +254,6 @@
 	$(PJLIB_LDLIB) \
 	@LIBS@
 export APP_LDXXLIBS := $(PJSUA2_LIB_LDLIB) \
-	-lstdc++ \
 	$(APP_LDLIBS)
 
 # Here are the variabels to use if application is using the library
diff -aur pjproject-2.2.1.orig/pjlib/src/pj/os_core_unix.c pjproject-2.2.1/pjlib/src/pj/os_core_unix.c
--- pjproject-2.2.1.orig/pjlib/src/pj/os_core_unix.c	2015-01-12 02:52:08.088675237 +0300
+++ pjproject-2.2.1/pjlib/src/pj/os_core_unix.c	2015-01-12 02:45:47.478653534 +0300
@@ -994,7 +994,8 @@
 PJ_DEF(pj_status_t) pj_thread_local_alloc(long *p_index)
 {
 #if PJ_HAS_THREADS
-    pthread_key_t key;
+    static int key_idx = 0;
+    pthread_key_t key = key_idx++;
     int rc;
 
     PJ_ASSERT_RETURN(p_index != NULL, PJ_EINVAL);
diff -aur pjproject-2.2.1.orig/pjlib/src/pj/sock_bsd.c pjproject-2.2.1/pjlib/src/pj/sock_bsd.c
--- pjproject-2.2.1.orig/pjlib/src/pj/sock_bsd.c	2015-01-12 02:52:08.092008571 +0300
+++ pjproject-2.2.1/pjlib/src/pj/sock_bsd.c	2015-01-12 02:45:47.478653534 +0300
@@ -52,7 +52,11 @@
 const pj_uint16_t PJ_SOCK_STREAM= SOCK_STREAM;
 const pj_uint16_t PJ_SOCK_DGRAM	= SOCK_DGRAM;
 const pj_uint16_t PJ_SOCK_RAW	= SOCK_RAW;
+#if defined(SOCK_RDM)
 const pj_uint16_t PJ_SOCK_RDM	= SOCK_RDM;
+#else
+const pj_uint16_t PJ_SOCK_RDM	= 0xFFFF;
+#endif
 
 /*
  * Socket level values.
diff -aur pjproject-2.2.1.orig/pjsip/build/Makefile pjproject-2.2.1/pjsip/build/Makefile
--- pjproject-2.2.1.orig/pjsip/build/Makefile	2015-01-12 02:52:08.125341908 +0300
+++ pjproject-2.2.1/pjsip/build/Makefile	2015-01-12 02:47:55.268660821 +0300
@@ -164,7 +164,7 @@
 			   main.o
 export PJSUA2_TEST_CFLAGS += $(_CFLAGS) $(PJ_VIDEO_CFLAGS)
 export PJSUA2_TEST_CXXFLAGS = $(PJSUA2_LIB_CFLAGS) 
-export PJSUA2_TEST_LDFLAGS += $(PJ_LDXXFLAGS) $(PJ_LDXXLIBS) $(LDFLAGS)
+export PJSUA2_TEST_LDFLAGS += 
 export PJSUA2_TEST_EXE := pjsua2-test-$(TARGET_NAME)$(HOST_EXE)
 
 export CC_OUT CC AR RANLIB HOST_MV HOST_RM HOST_RMDIR HOST_MKDIR OBJEXT LD LDOUT 
