diff -aur pjproject-2.10/Makefile ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/Makefile
--- pjproject-2.10/Makefile	2020-02-14 12:48:27.000000000 +0300
+++ ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/Makefile	2021-01-04 13:48:24.702794506 +0300
@@ -118,8 +118,8 @@
 
 install:
 	mkdir -p $(DESTDIR)$(libdir)/
-#	cp -af $(APP_LIB_FILES) $(DESTDIR)$(libdir)/
-	cp -af $(APP_LIBXX_FILES) $(DESTDIR)$(libdir)/
+	cp -af $(APP_LIB_FILES) $(DESTDIR)$(libdir)/
+#	cp -af $(APP_LIBXX_FILES) $(DESTDIR)$(libdir)/
 	mkdir -p $(DESTDIR)$(includedir)/
 	for d in pjlib pjlib-util pjnath pjmedia pjsip; do \
 		cp -RLf $$d/include/* $(DESTDIR)$(includedir)/; \
diff -aur pjproject-2.10/pjsip/build/Makefile ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/pjsip/build/Makefile
--- pjproject-2.10/pjsip/build/Makefile	2021-01-04 13:49:14.023213090 +0300
+++ ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/pjsip/build/Makefile	2021-01-04 13:30:25.734595581 +0300
@@ -209,9 +209,8 @@
 TARGETS := $(PJSIP_LIB) $(PJSIP_SONAME) \
 	   $(PJSIP_SIMPLE_LIB) $(PJSIP_SIMPLE_SONAME) \
 	   $(PJSIP_UA_LIB) $(PJSIP_UA_SONAME) \
-	   $(PJSUA_LIB_LIB) $(PJSUA_LIB_SONAME) \
-	   $(PJSUA2_LIB_LIB) $(PJSUA2_LIB_SONAME)
-TARGETS_EXE := $(TEST_EXE) $(PJSUA2_TEST_EXE)
+	   $(PJSUA_LIB_LIB) $(PJSUA_LIB_SONAME)
+TARGETS_EXE :=
 
 all: $(TARGETS) $(TARGETS_EXE)
 
@@ -321,9 +320,7 @@
 	$(MAKE) -f $(RULES_MAK) APP=PJSIP_UA app=pjsip-ua $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSIP_SIMPLE app=pjsip-simple $@
 	$(MAKE) -f $(RULES_MAK) APP=PJSUA_LIB app=pjsua-lib $@
-	$(MAKE) -f $(RULES_MAK) APP=PJSUA2_LIB app=pjsua2-lib $@
 	$(MAKE) -f $(RULES_MAK) APP=TEST app=pjsip-test $@
-	$(MAKE) -f $(RULES_MAK) APP=PJSUA2_TEST app=pjsua2-test $@
 	echo '$(BINDIR)/$(TEST_EXE): $(PJMEDIA_LIB) $(LIBDIR)/$(PJSUA_LIB_LIB) $(LIBDIR)/$(PJSIP_SIMPLE_LIB) $(LIBDIR)/$(PJSIP_UA_LIB) $(LIBDIR)/$(PJSIP_LIB) $(PJNATH_LIB) $(PJLIB_UTIL_LIB) $(PJLIB_LIB)' >> .pjsip-test-$(TARGET_NAME).depend
 	echo '$(BINDIR)/$(PJSUA2_TEST_EXE): $(LIBDIR)/$(PJSUA2_LIB_LIB) $(PJMEDIA_LIB) $(LIBDIR)/$(PJSUA_LIB_LIB) $(LIBDIR)/$(PJSIP_SIMPLE_LIB) $(LIBDIR)/$(PJSIP_UA_LIB) $(LIBDIR)/$(PJSIP_LIB) $(PJNATH_LIB) $(PJLIB_UTIL_LIB) $(PJLIB_LIB)' >> .pjsua2-test-$(TARGET_NAME).depend
 
diff -aur pjproject-2.10/pjsip-apps/build/Samples.mak ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/pjsip-apps/build/Samples.mak
--- pjproject-2.10/pjsip-apps/build/Samples.mak	2020-02-14 12:48:27.000000000 +0300
+++ ../build/extbld/third_party/pjproject/core_c/pjproject-2.10/pjsip-apps/build/Samples.mak	2021-01-04 13:30:25.738595587 +0300
@@ -58,7 +58,7 @@
 .PHONY: $(EXES)
 .PHONY: $(PJSUA2_EXES)
 
-all: $(EXES) $(PJSUA2_EXES)
+all: $(EXES)
 
 $(EXES):
 	$(MAKE) --no-print-directory -f $(RULES_MAK) SAMPLE_SRCDIR=$(SRCDIR) SAMPLE_OBJS=$@.o SAMPLE_CFLAGS="$(_CFLAGS)" SAMPLE_CXXFLAGS="$(_CXXFLAGS)" SAMPLE_LDFLAGS="$(_LDFLAGS)" SAMPLE_EXE=$@ APP=SAMPLE app=sample $(subst /,$(HOST_PSEP),$(BINDIR)/$@)
