diff -aur ./gcc-releases-gcc-13.3.0/gcc/Makefile.in ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/Makefile.in
--- ./gcc-releases-gcc-13.3.0/gcc/Makefile.in	2024-05-21 10:47:37.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/Makefile.in	2025-01-13 16:14:08.006937485 +0300
@@ -439,17 +439,7 @@
 
 # Header files that are made available under the same name
 # to programs compiled with GCC.
-USER_H = $(srcdir)/ginclude/float.h \
-	 $(srcdir)/ginclude/iso646.h \
-	 $(srcdir)/ginclude/stdarg.h \
-	 $(srcdir)/ginclude/stdbool.h \
-	 $(srcdir)/ginclude/stddef.h \
-	 $(srcdir)/ginclude/varargs.h \
-	 $(srcdir)/ginclude/stdfix.h \
-	 $(srcdir)/ginclude/stdnoreturn.h \
-	 $(srcdir)/ginclude/stdalign.h \
-	 $(srcdir)/ginclude/stdatomic.h \
-	 $(EXTRA_HEADERS)
+USER_H = $(EXTRA_HEADERS)
 
 USER_H_INC_NEXT_PRE = @user_headers_inc_next_pre@
 USER_H_INC_NEXT_POST = @user_headers_inc_next_post@
diff -aur ./gcc-releases-gcc-13.3.0/gcc/c/Make-lang.in ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/c/Make-lang.in
--- ./gcc-releases-gcc-13.3.0/gcc/c/Make-lang.in	2024-05-21 10:47:38.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/c/Make-lang.in	2025-01-13 16:05:38.850444369 +0300
@@ -73,14 +73,14 @@
 # the bootstrap comparison and allows comparing of the cc1 binary
 cc1-checksum.cc : build/genchecksum$(build_exeext) checksum-options \
 	$(C_OBJS) $(BACKEND) $(LIBDEPS) 
-	if [ -f ../stage_final ] \
-	   && cmp -s ../stage_current ../stage_final; then \
-	  cp ../prev-gcc/cc1-checksum.cc cc1-checksum.cc; \
-	else \
-	  build/genchecksum$(build_exeext) $(C_OBJS) $(BACKEND) $(LIBDEPS) \
-                     checksum-options > cc1-checksum.cc.tmp && 		 \
-	  $(srcdir)/../move-if-change cc1-checksum.cc.tmp cc1-checksum.cc; \
-	fi
+	#if [ -f ../stage_final ] \
+	#   && cmp -s ../stage_current ../stage_final; then \
+	#  cp ../prev-gcc/cc1-checksum.cc cc1-checksum.cc; \
+	#else \
+	#  build/genchecksum$(build_exeext) $(C_OBJS) $(BACKEND) $(LIBDEPS) \
+    #                 checksum-options > cc1-checksum.cc.tmp && 		 \
+	#  $(srcdir)/../move-if-change cc1-checksum.cc.tmp cc1-checksum.cc; \
+	#fi
 
 cc1$(exeext): $(C_OBJS) cc1-checksum.o $(BACKEND) $(LIBDEPS)
 	@$(call LINK_PROGRESS,$(INDEX.c),start)
diff -aur ./gcc-releases-gcc-13.3.0/gcc/cp/Make-lang.in ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/cp/Make-lang.in
--- ./gcc-releases-gcc-13.3.0/gcc/cp/Make-lang.in	2024-05-21 10:47:38.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/gcc/cp/Make-lang.in	2025-01-13 16:05:38.850444369 +0300
@@ -131,14 +131,14 @@
 # the bootstrap comparison and allows comparing of the cc1 binary
 cc1plus-checksum.cc : build/genchecksum$(build_exeext) checksum-options \
 	$(CXX_OBJS) $(BACKEND) $(CODYLIB) $(LIBDEPS) 
-	if [ -f ../stage_final ] \
-	   && cmp -s ../stage_current ../stage_final; then \
-	   cp ../prev-gcc/cc1plus-checksum.cc cc1plus-checksum.cc; \
-	else \
-	  build/genchecksum$(build_exeext) $(CXX_OBJS) $(BACKEND) $(CODYLIB) $(LIBDEPS) \
-                     checksum-options > cc1plus-checksum.cc.tmp &&	   \
-	  $(srcdir)/../move-if-change cc1plus-checksum.cc.tmp cc1plus-checksum.cc; \
-	fi
+	#if [ -f ../stage_final ] \
+	#   && cmp -s ../stage_current ../stage_final; then \
+	#   cp ../prev-gcc/cc1plus-checksum.cc cc1plus-checksum.cc; \
+	#else \
+	#  build/genchecksum$(build_exeext) $(CXX_OBJS) $(BACKEND) $(CODYLIB) $(LIBDEPS) \
+    #                 checksum-options > cc1plus-checksum.cc.tmp &&	   \
+	#  $(srcdir)/../move-if-change cc1plus-checksum.cc.tmp cc1plus-checksum.cc; \
+	#fi
 
 cc1plus$(exeext): $(CXX_OBJS) cc1plus-checksum.o $(BACKEND) $(CODYLIB) $(LIBDEPS) $(c++.prev)
 	@$(call LINK_PROGRESS,$(INDEX.c++),start)
diff -aur ./gcc-releases-gcc-13.3.0/libgcc/configure ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libgcc/configure
--- ./gcc-releases-gcc-13.3.0/libgcc/configure	2024-05-21 10:47:41.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libgcc/configure	2025-01-13 16:05:38.850444369 +0300
@@ -5048,6 +5048,9 @@
 target_thread_file=`$CC -v 2>&1 | sed -n 's/^Thread model: //p'`
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $target_thread_file" >&5
 $as_echo "$target_thread_file" >&6; }
+# In Embox we provide posix even if gcc built with single thread model
+# Among other solutions is to use Linux gcc, or our own gcc.
+target_thread_file=posix
 
 # Check for assembler CFI support.
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether assembler supports CFI directives" >&5
diff -aur ./gcc-releases-gcc-13.3.0/libstdc++-v3/configure ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/configure
--- ./gcc-releases-gcc-13.3.0/libstdc++-v3/configure	2024-05-21 10:47:42.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/configure	2025-01-13 16:05:38.858444439 +0300
@@ -15812,6 +15812,9 @@
   target_thread_file=`$CXX -v 2>&1 | sed -n 's/^Thread model: //p'`
   { $as_echo "$as_me:${as_lineno-$LINENO}: result: $target_thread_file" >&5
 $as_echo "$target_thread_file" >&6; }
+  # In Embox we provide posix even if gcc built with single thread model
+  # Among other solutions is to use Linux gcc, or our own gcc.
+  target_thread_file=posix
 
 case $target_thread_file in
     aix)	thread_header=config/rs6000/gthr-aix.h ;;
@@ -68277,7 +68280,7 @@
 
     ;;
   *)
-    as_fn_error $? "No support for this host/target combination." "$LINENO" 5
+    #as_fn_error $? "No support for this host/target combination." "$LINENO" 5
    ;;
 esac
 
@@ -69772,7 +69775,9 @@
   CXXFLAGS="$CXXFLAGS -fno-exceptions \
 	-I${toplevel_srcdir}/libgcc -I${toplevel_builddir}/libgcc"
 
-  target_thread_file=`$CXX -v 2>&1 | sed -n 's/^Thread model: //p'`
+  # In Embox we provide posix even if gcc built with single thread model
+  # Among other solutions is to use Linux gcc, or our own gcc.
+  target_thread_file=posix
   case $target_thread_file in
     posix)
       CXXFLAGS="$CXXFLAGS -DSUPPORTS_WEAK -DGTHREAD_USE_WEAK -D_PTHREADS"
@@ -73087,11 +73092,6 @@
       glibcxx_toolexecdir='${libdir}/gcc/${host_alias}'
       glibcxx_toolexeclibdir='${libdir}'
     fi
-    multi_os_directory=`$CXX -print-multi-os-directory`
-    case $multi_os_directory in
-      .) ;; # Avoid trailing /.
-      *) glibcxx_toolexeclibdir=$glibcxx_toolexeclibdir/$multi_os_directory ;;
-    esac
   fi
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for install location" >&5
diff -aur ./gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.am ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.am
--- ./gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.am	2024-05-21 10:47:42.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.am	2025-01-13 16:05:38.858444439 +0300
@@ -133,8 +133,6 @@
 	locale_init.cc \
 	locale_facets.cc \
 	localename.cc \
-	math_stubs_float.cc \
-	math_stubs_long_double.cc \
 	stdexcept.cc \
 	strstream.cc \
 	tree.cc \
diff -aur ./gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.in ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.in
--- ./gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.in	2024-05-21 10:47:42.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/libstdc++-v3/src/c++98/Makefile.in	2025-01-13 16:05:38.858444439 +0300
@@ -140,7 +140,7 @@
 	hashtable_tr1.lo ios_failure.lo ios_init.lo ios_locale.lo \
 	list.lo list-aux.lo list-aux-2.lo list_associated.lo \
 	list_associated-2.lo locale.lo locale_init.lo locale_facets.lo \
-	localename.lo math_stubs_float.lo math_stubs_long_double.lo \
+	localename.lo \
 	stdexcept.lo strstream.lo tree.lo istream.lo istream-string.lo \
 	streambuf.lo valarray.lo $(am__objects_1) $(am__objects_3) \
 	$(am__objects_6)
@@ -509,8 +509,6 @@
 	locale_init.cc \
 	locale_facets.cc \
 	localename.cc \
-	math_stubs_float.cc \
-	math_stubs_long_double.cc \
 	stdexcept.cc \
 	strstream.cc \
 	tree.cc \
diff -aur ./gcc-releases-gcc-13.3.0/zlib/Makefile.in ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/zlib/Makefile.in
--- ./gcc-releases-gcc-13.3.0/zlib/Makefile.in	2024-05-21 10:49:00.000000000 +0300
+++ ../build/extbld/third_party/gcc/gcc_build/gcc-releases-gcc-13.3.0/zlib/Makefile.in	2025-01-13 16:05:38.858444439 +0300
@@ -452,7 +452,7 @@
 MULTISUBDIR = 
 MULTIDO = true
 MULTICLEAN = true
-all: all-am
+all:
 
 .SUFFIXES:
 .SUFFIXES: .c .lo .o .obj
