--- framework-k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h
+++ ../build/extbld/third_party/bsp/k1921vk_sdk/k1921vk035_system/framework-k1921vk-sdk-1.7.3/platform/Device/NIIET/K1921VK035/Include/system_K1921VK035.h
@@ -27,6 +27,14 @@
 
 //-- Includes ------------------------------------------------------------------
 #include <stdint.h>
+#ifdef __EMBOX__
+#include <framework/mod/options.h>
+#include <module/embox/arch/system.h>
+#include <module/third_party/bsp/k1921vk_sdk/k1921vk035_system.h>
+
+#define OSECLK_VAL OPTION_MODULE_GET(third_party__bsp__k1921vk_sdk__k1921vk035_system,NUMBER,external_clk_freq)
+#define SYSCLK_PLL
+#endif
 
 //-- Defines -------------------------------------------------------------------
 #define OSICLK_VAL 8000000
