diff -aur dropbear-2022.83/Makefile.in ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/Makefile.in
--- dropbear-2022.83/Makefile.in	2022-11-14 20:30:00.000000000 +0600
+++ ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/Makefile.in	2023-07-17 15:08:55.831321280 +0600
@@ -12,6 +12,8 @@
 	PROGRAMS=dropbear dbclient dropbearkey dropbearconvert
 endif
 
+CPPFLAGS+=-DDROPBEAR_EMBOX_BUILD
+
 STATIC_LTC=libtomcrypt/libtomcrypt.a
 STATIC_LTM=libtommath/libtommath.a
 
diff -aur dropbear-2022.83/debug.h ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/debug.h
--- dropbear-2022.83/debug.h	2022-11-14 20:30:00.000000000 +0600
+++ ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/debug.h	2023-07-17 15:00:27.728009909 +0600
@@ -89,7 +89,7 @@
 /* To debug with GDB it is easier to run with no forking of child processes.
    You will need to pass "-F" as well. */
 #ifndef DEBUG_NOFORK
-#define DEBUG_NOFORK 0
+#define DEBUG_NOFORK 1
 #endif
 
 
diff -aur dropbear-2022.83/includes.h ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/includes.h
--- dropbear-2022.83/includes.h	2022-11-14 20:30:00.000000000 +0600
+++ ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/includes.h	2023-07-17 15:00:27.728009909 +0600
@@ -28,9 +28,15 @@
 #include "options.h"
 #include "debug.h"
 
+#ifdef DROPBEAR_EMBOX_BUILD
+#include "stubs.h"
+#endif
+
 #include <sys/types.h>
 #include <sys/ioctl.h>
+#ifndef DROPBEAR_EMBOX_BUILD
 #include <sys/param.h> /* required for BSD4_4 define */
+#endif
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/time.h>
@@ -49,7 +55,9 @@
 #include <string.h>
 #include <termios.h>
 #include <unistd.h>
+#ifndef DROPBEAR_EMBOX_BUILD
 #include <syslog.h>
+#endif
 #include <netdb.h>
 #include <ctype.h>
 #include <stdarg.h>
@@ -85,7 +93,9 @@
 #include <netinet/in_systm.h>
 #endif
 
+#ifndef DROPBEAR_EMBOX_BUILD
 #include <netinet/ip.h>
+#endif
 
 #ifdef HAVE_NETINET_TCP_H
 #include <netinet/tcp.h>
diff -aur dropbear-2022.83/sysoptions.h ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/sysoptions.h
--- dropbear-2022.83/sysoptions.h	2022-11-14 20:30:00.000000000 +0600
+++ ../build/extbld/third_party/dropbear/dropbear/dropbear-2022.83/sysoptions.h	2023-07-17 15:00:27.728009909 +0600
@@ -75,7 +75,11 @@
 /* Each port might have at least a v4 and a v6 address */
 #define MAX_LISTEN_ADDR (DROPBEAR_MAX_PORTS*3)
 
+#ifndef DROPBEAR_EMBOX_BUILD
 #define _PATH_TTY "/dev/tty"
+#else
+#define _PATH_TTY "/dev_pty_0"
+#endif
 
 #define _PATH_CP "/bin/cp"
 
@@ -299,9 +303,13 @@
 
 /* Source for randomness. This must be able to provide hundreds of bytes per SSH
  * connection without blocking. */
+#ifndef DROPBEAR_EMBOX_BUILD
 #ifndef DROPBEAR_URANDOM_DEV
 #define DROPBEAR_URANDOM_DEV "/dev/urandom"
 #endif
+#else
+#define DROPBEAR_URANDOM_DEV "/etc/dropbear/random"
+#endif
 
 /* client keyboard interactive authentication is often used for password auth.
  rfc4256 */
