diff -aur libmodbus-3.1.6/configure ../build/extbld/third_party/lib/libmodbus/libmodbus-3.1.6/configure
--- libmodbus-3.1.6/configure	2019-08-01 00:38:45.169833195 +0300
+++ ../build/extbld/third_party/lib/libmodbus/libmodbus-3.1.6/configure	2021-02-18 16:14:14.550602956 +0300
@@ -13527,7 +13527,7 @@
 
 
 # Checks for library functions.
-for ac_func in accept4 getaddrinfo gettimeofday inet_ntoa select socket strerror strlcpy
+for ac_func in getaddrinfo gettimeofday inet_ntoa select socket strerror strlcpy
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"